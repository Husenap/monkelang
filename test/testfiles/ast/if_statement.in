monke think bananas is greater than 10 monke do
  monke say 9999
monke think again bananas is greater than 5 monke do
  monke say 10
monke think again bananas is equal to 5 monke do
  monke say 5
monke change mind
  monke say 0
monke thinked


monke think bananas is greater than 10 monke do
  monke say 9999
monke think again bananas is equal to 5 monke do
  monke say 5
monke change mind
  monke say 0
monke thinked


monke think bananas is greater than 10 monke do
  monke say 9999
monke change mind
  monke say 0
monke thinked


monke think bananas is greater than 10 monke do
  monke say 9999
monke thinked


