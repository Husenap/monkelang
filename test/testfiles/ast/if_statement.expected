IfStmt: 
  Condition: 
    BinOpExpr: >
      IdUseExpr: bananas
      NumberExpr: 10
  Body: 
    PrintStmt: 
      NumberExpr: 9999
  Else-If Branch: 
    Condition: 
    BinOpExpr: >
      IdUseExpr: bananas
      NumberExpr: 5
    Body: 
    PrintStmt: 
      NumberExpr: 10
  Else-If Branch: 
    Condition: 
    BinOpExpr: ==
      IdUseExpr: bananas
      NumberExpr: 5
    Body: 
    PrintStmt: 
      NumberExpr: 5
  Else Branch: 
    PrintStmt: 
      NumberExpr: 0
IfStmt: 
  Condition: 
    BinOpExpr: >
      IdUseExpr: bananas
      NumberExpr: 10
  Body: 
    PrintStmt: 
      NumberExpr: 9999
  Else-If Branch: 
    Condition: 
    BinOpExpr: ==
      IdUseExpr: bananas
      NumberExpr: 5
    Body: 
    PrintStmt: 
      NumberExpr: 5
  Else Branch: 
    PrintStmt: 
      NumberExpr: 0
IfStmt: 
  Condition: 
    BinOpExpr: >
      IdUseExpr: bananas
      NumberExpr: 10
  Body: 
    PrintStmt: 
      NumberExpr: 9999
  Else Branch: 
    PrintStmt: 
      NumberExpr: 0
IfStmt: 
  Condition: 
    BinOpExpr: >
      IdUseExpr: bananas
      NumberExpr: 10
  Body: 
    PrintStmt: 
      NumberExpr: 9999
